Errors for Graphene samples

### Issue description
Trying the graphene samples hello and nginx fails with an error message.

### To reproduce
Steps to reproduce the behavior:

1. Build and install Graphene with version b37ac75e
2. Follow the `graphene-hello` and `graphen-nginx` samples.

### Environment:
 - Marblerun version: 0.4.0
 - Edgeless RT version: 0.2.5
 - Graphene version: b37ac75e

### Additional info / screenshots
```
$ EDG_MARBLE_TYPE=hello \
EDG_MARBLE_COORDINATOR_ADDR=localhost:2001 \
EDG_MARBLE_UUID_FILE=uuid \
EDG_MARBLE_DNS_NAMES=localhost \
make run
graphene-sgx hello
error: Forwarding host environment variables to the app is enabled. Graphene will continue application execution, but this configuration must not be used in production!
[P29462:T1:] error: 'libos.entrypoint' is now a Graphene path, not URI. Ignoring the 'file:' prefix.
[P29462:T1:] error: init_exec_handle: error opening executable: -13
[P29462:T1:] error: Error during shim_init() in init_important_handles (-13)
Makefile:29: recipe for target 'run' failed
make: *** [run] Error 13
```

```
$ EDG_MARBLE_TYPE=frontend \
EDG_MARBLE_COORDINATOR_ADDR=localhost:2001 \
EDG_MARBLE_UUID_FILE=uuid \
EDG_MARBLE_DNS_NAMES=localhost \
graphene-sgx nginx
error: Using insecure argv source. Graphene will continue application execution, but this configuration must not be used in production!
error: Forwarding host environment variables to the app is enabled. Graphene will continue application execution, but this configuration must not be used in production!
[P32530:T1:] error: 'libos.entrypoint' is now a Graphene path, not URI. Ignoring the 'file:' prefix.
[P32530:T1:] error: init_exec_handle: error opening executable: -13
[P32530:T1:] error: Error during shim_init() in init_important_handles (-13)
```

--- a/cli/cmd/graphenePrepare.go
+++ b/cli/cmd/graphenePrepare.go
@@ -102,26 +102,18 @@ func parseTreeForChanges(tree *toml.Tree) (map[string]interface{}, map[string]in
 	}
 
 	// If Marblerun already touched the manifest, abort.
-	if original["libos.entrypoint"].(string) == "file:"+premainName || original["sgx.trusted_files.marblerun_premain"] != nil || original["sgx.allowed_files.marblerun_uuid"] != nil {
+	if original["libos.entrypoint"].(string) == premainName || original["sgx.trusted_files.marblerun_premain"] != nil || original["sgx.allowed_files.marblerun_uuid"] != nil {
 		color.Yellow("The supplied manifest already contains changes for Marblerun. Have you selected the correct file?")
 		return nil, nil, errors.New("manifest already contains Marblerun changes")
 	}
 
 	// Add premain-libos executable as trusted file & entry point
-	changes["libos.entrypoint"] = "file:" + premainName
+	changes["libos.entrypoint"] = premainName
 	changes["sgx.trusted_files.marblerun_premain"] = "file:" + premainName
 
 	// Set original endpoint as argv0. If one exists, keep the old one
 	if original["loader.argv0_override"] == nil {
-		fileEntry := strings.SplitAfter(original["libos.entrypoint"].(string), "file:")
-		if len(fileEntry) != 2 {
-			color.Red("ERROR: Cannot process the current entrypoint: %s", original["libos.entrypoint"].(string))
-			color.Red("Note: This tool only supports 'file:' URIs for automatic modifcation.")
-			color.Red("If you chose another type of path reference, please change it to 'file:' to continue.")
-			color.Red("Otherwise, please file a bug report!")
-			return nil, nil, fmt.Errorf("cannot determine entrypoint for argv0 override correctly")
-		}
-		changes["loader.argv0_override"] = fileEntry[1]
+		changes["loader.argv0_override"] = original["libos.entrypoint"].(string)
 	}
 
 	// Enable use "insecure" host env (which delegates the "secure" handling to Marblerun)

--- a/cli/cmd/graphenePrepare_test.go
+++ b/cli/cmd/graphenePrepare_test.go
@@ -15,7 +15,7 @@ import (
 )
 
 const someManifest = `
-libos.entrypoint = "file:myapplication"
+libos.entrypoint = "myapplication"
 sgx.remote_attestation = 0
 # Some comment here in between
 # This should not match: sgx.enclave_size - 2
@@ -71,7 +71,7 @@ func TestParseTreeForChanges(t *testing.T) {
 	// Verify minimum changes
 	var v datasize.ByteSize
 
-	assert.Equal("file:"+premainName, changes["libos.entrypoint"])
+	assert.Equal(premainName, changes["libos.entrypoint"])
 	assert.GreaterOrEqual(changes["sgx.thread_num"], 16)
 	require.NoError(v.UnmarshalText([]byte(changes["sgx.enclave_size"].(string))))
 	assert.GreaterOrEqual(v.GBytes(), 1.00)

--- a/samples/.gitignore
+++ b/samples/.gitignore
@@ -1,0 +1,10 @@
+*.token
+*.sgx
+uuid
+*.tar.gz
+*.sig
+premain-libos
+install/
+nginx-1.16.1/
+*.manifest
+hello

--- a/samples/graphene-hello/Makefile
+++ b/samples/graphene-hello/Makefile
@@ -23,7 +23,7 @@ hello.manifest: hello.manifest.template
 
 premain-libos:
 	wget https://github.com/edgelesssys/marblerun/releases/latest/download/premain-libos
-
+	chmod u+x premain-libos
 
 run:
 	graphene-sgx hello

--- a/samples/graphene-hello/hello.manifest.template
+++ b/samples/graphene-hello/hello.manifest.template
@@ -2,7 +2,7 @@ loader.preload = "file:{{ graphene.libos }}"
 loader.env.LD_LIBRARY_PATH = "/lib"
 
 # entrypoint must be premain-libos
-libos.entrypoint = "file:premain-libos"
+libos.entrypoint = "premain-libos"
 sgx.trusted_files.premain = "file:premain-libos"
 
 # argv0 must be the path to the actual application

--- a/samples/graphene-hello/hello.manifest.template
+++ b/samples/graphene-hello/hello.manifest.template
@@ -35,9 +35,13 @@ fs.mount.etc.type = "chroot"
 fs.mount.etc.path = "/etc"
 fs.mount.etc.uri = "file:/etc"
 sgx.trusted_files.nss = "file:{{ graphene.runtimedir() }}/libnss_dns.so.2"
+sgx.trusted_files.nss_files = "file:{{ graphene.runtimedir() }}/libnss_files.so.2"
 sgx.trusted_files.resolv = "file:{{ graphene.runtimedir() }}/libresolv.so.2"
 sgx.allowed_files.hosts = "file:/etc/hosts"
-sgx.allowed_files.resolv = "file:/etc/resolv.conf"
+sgx.allowed_files.host_conf = "file:/etc/host.conf"
+sgx.allowed_files.gai_conf = "file:/etc/gai.conf"
+sgx.allowed_files.resolv_conf = "file:/etc/resolv.conf"
+sgx.allowed_files.localtime = "file:/etc/localtime"
 
 
 # Name Service Switch (NSS) files, see 'man nsswitch.conf'

--- a/samples/graphene-nginx/Makefile
+++ b/samples/graphene-nginx/Makefile
@@ -73,12 +73,17 @@ nginx.manifest: nginx.manifest.template
 		-Dinstall_dir_abspath=$(abspath $(INSTALL_DIR)) \
 		$< >$@
 
+premain-libos:
+	wget https://github.com/edgelesssys/marblerun/releases/latest/download/premain-libos
+	chmod u+x premain-libos
+
 # Generating the SGX-specific manifest (nginx.manifest.sgx), the enclave signature,
 # and the token for enclave initialization.
 nginx.manifest.sgx: nginx.manifest $(INSTALL_DIR)/sbin/nginx \
 		$(INSTALL_DIR)/conf/nginx-graphene.conf \
 		$(TEST_DATA) \
-		$(INSTALL_DIR)/conf/server.crt
+		$(INSTALL_DIR)/conf/server.crt \
+		premain-libos
 	graphene-sgx-sign \
 		--key $(SGX_SIGNER_KEY) \
 		--manifest $< \

--- a/samples/graphene-nginx/README.md
+++ b/samples/graphene-nginx/README.md
@@ -11,7 +11,6 @@ We also removed certificate generation from the Makefile because it will be prov
 
 We now build the example as follows:
 ```sh
-wget https://github.com/edgelesssys/marblerun/releases/latest/download/premain-libos
 export GRAPHENEDIR=[PATH To Your Graphene Folder]
 make SGX=1
 ```

--- a/samples/graphene-nginx/nginx.manifest.template
+++ b/samples/graphene-nginx/nginx.manifest.template
@@ -3,7 +3,7 @@
 # This manifest was prepared and tested on Ubuntu 18.04.
 
 # MARBLERUN: entrypoint must be premain-libos
-libos.entrypoint = "file:premain-libos"
+libos.entrypoint = "premain-libos"
 
 # Path to the library OS
 loader.preload = "file:{{ graphene.libos }}"

